### ISSUE TITLE
Can't Save the Customer

### ISSUE THREAD

**Issue by InternetMedicineMan** [magepal/magento2-gmailsmtpapp#56]
So I had a problem that I thought was relating to other plugins. But customers can't save, and Pending Registrations won't process, all because of email being sent out. I thought it was because of this plugin first, but then when disabling something I thought that the problem went away, so I thought it was other 3rd party plugins. But it's not, after doing further research and disabling all the other 3rd party plugins I'm seeing the same error. And after tracking it down further I'm seeing it's directly relating to this plugin.

When trying to save a customer it keeps coming up with an error that it can't save the customer from the front end, or from the backend it was creating the customer, but showing an error about the email. 

When creating a customer I get this exception:

`
[2017-03-03 19:15:22] main.CRITICAL: Exception message: Warning: strtr() expects parameter 1 to be string, array given in /var/www/html/vendor/magento/zendframework1/library/Zend/Mail.php on line 1216
Trace: #0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2, 'strtr() expects...', '/var/www/html/v...', 1216, Array)
#1 /var/www/html/vendor/magento/zendframework1/library/Zend/Mail.php(1216): strtr(Array, Array)
#2 /var/www/html/vendor/magento/zendframework1/library/Zend/Mail.php(683): Zend_Mail->_filterEmail(Array)
#3 /var/www/html/vendor/magepal/magento2-gmailsmtpapp/Model/Transport.php(58): Zend_Mail->setFrom(Array)
#4 /var/www/html/var/generation/MagePal/GmailSmtpApp/Model/Transport/Interceptor.php(14): MagePal\GmailSmtpApp\Model\Transport->__construct(Object(Magento\Framework\Mail\Message), Object(MagePal\GmailSmtpApp\Helper\Data))
#5 /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(93): MagePal\GmailSmtpApp\Model\Transport\Interceptor->__construct(Object(Magento\Framework\Mail\Message), Object(MagePal\GmailSmtpApp\Helper\Data))
#6 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(89): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('MagePal\\GmailSm...', Array)
#7 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('MagePal\\GmailSm...', Array)
#8 /var/www/html/vendor/magento/framework/Mail/TransportInterfaceFactory.php(50): Magento\Framework\ObjectManager\ObjectManager->create('Magento\\Framewo...', Array)
#9 /var/www/html/vendor/magento/framework/Mail/Template/TransportBuilder.php(228): Magento\Framework\Mail\TransportInterfaceFactory->create(Array)
#10 /var/www/html/vendor/magento/module-customer/Model/EmailNotification.php(237): Magento\Framework\Mail\Template\TransportBuilder->getTransport()
#11 /var/www/html/vendor/magento/module-customer/Model/EmailNotification.php(363): Magento\Customer\Model\EmailNotification->sendEmailTemplate(Object(Magento\Customer\Model\Data\Customer), 'customer/create...', 'customer/create...', Array, 1)
#12 /var/www/html/vendor/magento/module-customer/Model/AccountManagement.php(777): Magento\Customer\Model\EmailNotification->newAccount(Object(Magento\Customer\Model\Data\Customer), 'registered', NULL, 1)
#13 /var/www/html/vendor/magento/module-customer/Model/AccountManagement.php(737): Magento\Customer\Model\AccountManagement->sendEmailConfirmation(Object(Magento\Customer\Model\Data\Customer), NULL)
#14 /var/www/html/var/generation/Magento/Customer/Model/AccountManagement/Interceptor.php(141): Magento\Customer\Model\AccountManagement->createAccountWithPasswordHash(Object(Magento\Customer\Model\Data\Customer), '2cea6e76ad552ab...', NULL)
#15 /var/www/html/vendor/magento/module-customer/Model/AccountManagement.php(662): Magento\Customer\Model\AccountManagement\Interceptor->createAccountWithPasswordHash(Object(Magento\Customer\Model\Data\Customer), '2cea6e76ad552ab...', NULL)
#16 /var/www/html/var/generation/Magento/Customer/Model/AccountManagement/Interceptor.php(128): Magento\Customer\Model\AccountManagement->createAccount(Object(Magento\Customer\Model\Data\Customer), 'Test1111', NULL)
#17 /var/www/html/vendor/magento/module-customer/Controller/Account/CreatePost.php(277): Magento\Customer\Model\AccountManagement\Interceptor->createAccount(Object(Magento\Customer\Model\Data\Customer), 'Test1111', NULL)
`

It seems to still be referencing this plugin.

**Comment by srenon** [comment]
@InternetMedicineMan ... can you provide me the steps, so I can try to reproduce it on my local box?

Are you running multi-store?

**Comment by InternetMedicineMan** [comment]
Yes I'm running Multi-store. And all I'm doing is creating customers. On the frontend if I register it says can't save customer, and that's the exception above. On the Backend it saves the customer but reports and error "Invalid Sender Data". 

When I had the other plugins turned on I was approving customers and the approval would fail because it triggered an email. But orders process fine and send and email. It just seems wrapped around the customers now. But with Amazon configured it was an issue with orders.

**Comment by InternetMedicineMan** [comment]
I'm a fool again. I made a change to the $message->setFrom on Line 54 for the Transport.php file trying to get the Amazon SES to work, and when I switched to Google Apps, I forgot to switch it back. When Orders are being sent out $returnPathEmail is null on my install so that code isn't processed which allows it to work, but when anything else is done it's not null, and it was failing because I had set an array. (which also explains why that change did nothing for Amazon SES). Sorry for wasting your time. (again)
